File: app/src/main/java/com/mapelli/simone/githubclient/UI/ProfileFragment.java
package com.mapelli.simone.githubclient.UI;


import android.os.Bundle;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.mapelli.simone.githubclient.R;
import com.mapelli.simone.githubclient.data.entity.UserProfile_Full;


public class ProfileFragment extends Fragment {
    private ImageView user_photo;
    private TextView  name, location, email, profile_url;
    private String    location_str,email_str;

    private UserDetailActivity parent;
    private UserProfile_Full   currentUser;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        View rootView = inflater.inflate(R.layout.fragment_profile, container, false);
        parent = (UserDetailActivity) getActivity();
        currentUser = parent.getCurrentUser();

        user_photo  = rootView.findViewById(R.id.user_photo_img);
        name        = rootView.findViewById(R.id.name_txt);
        location    = rootView.findViewById(R.id.location_txt);
        email       = rootView.findViewById(R.id.email_txt);
        profile_url = rootView.findViewById(R.id.profile_url_txt);


        Glide.with(parent.getBaseContext())
                .load(currentUser.getAvatar_url())
                .fitCenter()
                .into(user_photo);

        location_str = currentUser.getLocation();
        if (location == null) currentUser.setLocation("No location present");

        email_str = currentUser.getEmail();
        if (email_str == null) currentUser.setEmail("No email present");

        name.setText(currentUser.getName());
        location.setText(currentUser.getLocation());
        email.setText(currentUser.getEmail());
        profile_url.setText(currentUser.getRepos_url());


        return rootView;
    }

}

File: app/src/main/java/com/mapelli/simone/githubclient/UI/ProfileFragment.kt
package com.mapelli.simone.githubclient.UI


import android.os.Bundle
import android.util.Log
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.bumptech.glide.Glide
import com.mapelli.simone.githubclient.R
import com.mapelli.simone.githubclient.data.entity.UserProfile_Full


class ProfileFragment : Fragment() {
    lateinit var user_photo: ImageView
    lateinit var name: TextView
    lateinit var location: TextView
    lateinit var email: TextView
    lateinit var profile_url: TextView
    lateinit var location_str: String
    lateinit var email_str: String

    lateinit var parent: UserDetailActivity
    lateinit var currentUser: UserProfile_Full

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        Log.d(TAG, "CHECK : inside ProfileFragment")
        val rootView = inflater.inflate(R.layout.fragment_profile, container, false)
        parent = activity as UserDetailActivity
        currentUser = parent.currentUser

        user_photo = rootView.findViewById(R.id.user_photo_img)
        name = rootView.findViewById(R.id.name_txt)
        location = rootView.findViewById(R.id.location_txt)
        email = rootView.findViewById(R.id.email_txt)
        profile_url = rootView.findViewById(R.id.profile_url_txt)

        location_str = currentUser.location!!
        currentUser.location ?: "No location present"

        email_str = currentUser.email!!
        currentUser.email ?: "No email present"

        name.text = currentUser.name
        location.text = currentUser.location
        email.text = currentUser.email
        profile_url.text = currentUser.repos_url

        Glide.with(parent.baseContext)
                .load(currentUser.avatar_url)
                .fitCenter()
                .into(user_photo)


        return rootView
    }


    companion object {
        private val TAG = ProfileFragment::class.java.getSimpleName()
    }

}
